sycl/doc/extensions/experimental/sycl_ext_oneapi_kernel_compiler.asciidoc

```diff
@@ -290,7 +309,78 @@ exception is caught and handled appropriately.
 _{endnote}_]
 |====
 
-=== New properties
+=== New properties for the `create_kernel_bundle_from_source` function
+
+This extension adds the following properties, which can be used in conjunction
+with the `create_kernel_bundle_from_source` function that is defined above:
+
+|====
+a|
+[frame=all,grid=none]
+!====
+a!
+[source]
+----
+namespace sycl::ext::oneapi::experimental {
+
+struct include_files {
+  std::vector<std::string> names;
+  std::vector<std::string> contents;
+  include_files(const std::string &name, const std::string &content);  (1)
+  include_files(const std::vector<std::string> &names,                 (2)
+                const std::vector<std::string> &contents);
+};
```

Comment:
I have some reservations about this include_files struct.

We know for the first implementation that we are likely going to have to write out a temporary .cpp file and invoke `clang++ -fsycl` on it.  So now we have to also write out the include files?  What about `#include "path/to/file.h"` ? Will we then be parsing this and instantiating directories?  

I presume we need to disambiguate `#include "local/path.h"` from `#include <system/path.h>` ?

Users can't link in libraries ( even if has a header ), and there is only the one TU.   Supporting `#include` suggests the opposite. Maybe a note proscribing it is needed?

I do see the value in that if a user has a simple function that they want to both use in the host application and call from their kernel, this allows them to do so.  And, if clang were to ever have an invocation API (like ocloc) then I guess this aligns with that.

Reply:
One of the reasons we added the `include_files` property is for parity with CUDA, which has a similar feature in NVRTC.  By adding the feature to SYCL, we make it easier to migrate code automatically using the SYCLomatic tool.

I was thinking that our implementation could create a temporary "include" directory and then write out the `include_files` header files into that directory.  When we invoke clang, we can add a `-I` switch pointing at the temporary include directory.

Note that you don't need to parse any `#include` strings from the source code to get the file name paths.  The `include_files` properties contains a list of (_Name_, _Content_) pairs.  Each _Name_ is the relative pathname of an include file, so you can just use this _Name_ string verbatim.